src/ekf.cpp

```diff
@@ -33,364 +33,422 @@
-    transfer_function_jacobian_(StateMemberX, StateMemberRoll) = dFx_dR;
-    transfer_function_jacobian_(StateMemberX, StateMemberPitch) = dFx_dP;
-    transfer_function_jacobian_(StateMemberX, StateMemberYaw) = dFx_dY;
-    transfer_function_jacobian_(StateMemberY, StateMemberRoll) = dFy_dR;
-    transfer_function_jacobian_(StateMemberY, StateMemberPitch) = dFy_dP;
-    transfer_function_jacobian_(StateMemberY, StateMemberYaw) = dFy_dY;
-    transfer_function_jacobian_(StateMemberZ, StateMemberRoll) = dFz_dR;
-    transfer_function_jacobian_(StateMemberZ, StateMemberPitch) = dFz_dP;
-    transfer_function_jacobian_(StateMemberRoll, StateMemberRoll) = dFR_dR;
-    transfer_function_jacobian_(StateMemberRoll, StateMemberPitch) = dFR_dP;
-    transfer_function_jacobian_(StateMemberRoll, StateMemberYaw) = dFR_dY;
-    transfer_function_jacobian_(StateMemberPitch, StateMemberRoll) = dFP_dR;
-    transfer_function_jacobian_(StateMemberPitch, StateMemberPitch) = dFP_dP;
-    transfer_function_jacobian_(StateMemberPitch, StateMemberYaw) = dFP_dY;
-    transfer_function_jacobian_(StateMemberYaw, StateMemberRoll) = dFY_dR;
-    transfer_function_jacobian_(StateMemberYaw, StateMemberPitch) = dFY_dP;
-
-    FB_DEBUG("Transfer function is:\n" << transfer_function_ <<
-             "\nTransfer function Jacobian is:\n" << transfer_function_jacobian_ <<
-             "\nProcess noise covariance is:\n" << process_noise_covariance_ <<
-             "\nCurrent state is:\n" << state_ << "\n");
-
-    Eigen::MatrixXd *process_noise_covariance = &process_noise_covariance_;
-
-    if (use_dynamic_process_noise_covariance_)
-    {
-      computeDynamicProcessNoiseCovariance(state_);
-      process_noise_covariance = &dynamic_process_noise_covariance_;
-    }
-
-    // (1) Apply control terms, which are actually accelerations
-    state_(StateMemberVroll) += control_acceleration_(ControlMemberVroll) * delta_sec;
-    state_(StateMemberVpitch) += control_acceleration_(ControlMemberVpitch) * delta_sec;
-    state_(StateMemberVyaw) += control_acceleration_(ControlMemberVyaw) * delta_sec;
-
-    state_(StateMemberAx) = (control_update_vector_[ControlMemberVx] ?
-      control_acceleration_(ControlMemberVx) : state_(StateMemberAx));
-    state_(StateMemberAy) = (control_update_vector_[ControlMemberVy] ?
-      control_acceleration_(ControlMemberVy) : state_(StateMemberAy));
-    state_(StateMemberAz) = (control_update_vector_[ControlMemberVz] ?
-      control_acceleration_(ControlMemberVz) : state_(StateMemberAz));
-
-    // (2) Project the state forward: x = Ax + Bu (really, x = f(x, u))
-    state_ = transfer_function_ * state_;
+    // (3) Apply the gain to the difference between the state and measurement: x
+    // = x + K(z - Hx)
+    state_.noalias() += kalman_gain_subset * innovation_subset;
+
+    // (4) Update the estimate error covariance using the Joseph form: (I -
+    // KH)P(I - KH)' + KRK'
+    Eigen::MatrixXd gainResidual = identity_;
```

Comment:
Nuts, looks like this one escaped my variable scheme fixing. Can you change this to `gain_residual`?

Reply:
Fixed in https://github.com/cra-ros-pkg/robot_localization/commit/56df59576707bb8ce6de3f9f2327622d07d112ab